php/class-customize-snapshot-manager.php

```diff
@@ -675,74 +298,17 @@ public function enqueue_controls_scripts() {
 					'save' => __( 'You do not have permission to publish changes, but you can create a snapshot by clicking the "Save" button.', 'customize-snapshots' ),
 					'update' => __( 'You do not have permission to publish changes, but you can modify this snapshot by clicking the "Update" button.', 'customize-snapshots' ),
 				),
+				'aysMsg' => __( 'Changes that you made may not be saved.', 'customize-snapshots' ),
 				'errorMsg' => __( 'The snapshot could not be saved.', 'customize-snapshots' ),
 				'errorTitle' => __( 'Error', 'customize-snapshots' ),
 				'collapseSnapshotScheduling' => __( 'Collapse snapshot scheduling', 'customize-snapshots' ),
 				'expandSnapshotScheduling' => __( 'Expand snapshot scheduling', 'customize-snapshots' ),
 			),
-			'snapshotExists' => ( $this->snapshot && $this->snapshot->saved() ),
 		) );
 
-		// Export data to JS.
-		wp_scripts()->add_data(
-			$this->plugin->slug,
-			'data',
-			sprintf( 'var _customizeSnapshots = %s;', wp_json_encode( $exports ) )
-		);
-	}
-
-	/**
-	 * Set up Customizer preview.
-	 */
-	public function customize_preview_init() {
-		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_preview_scripts' ) );
-	}
-
-	/**
-	 * Enqueue Customizer preview scripts.
-	 *
-	 * @global \WP_Customize_Manager $wp_customize
-	 */
-	public function enqueue_preview_scripts() {
-		global $wp_customize;
-
-		$handle = 'customize-snapshots-preview';
-		wp_enqueue_script( $handle );
-		wp_enqueue_style( $handle );
-
-		$exports = array(
-			'home_url' => wp_parse_url( home_url( '/' ) ),
-			'rest_api_url' => wp_parse_url( rest_url( '/' ) ),
-			'admin_ajax_url' => wp_parse_url( admin_url( 'admin-ajax.php' ) ),
-			'initial_dirty_settings' => array_keys( $wp_customize->unsanitized_post_values() ),
-		);
 		wp_add_inline_script(
-			$handle,
-			sprintf( 'CustomizeSnapshotsPreview.init( %s )', wp_json_encode( $exports ) ),
-			'after'
-		);
-	}
-
-	/**
-	 * Enqueue Customizer frontend scripts.
-	 */
-	public function enqueue_frontend_scripts() {
-		if ( ! $this->snapshot || is_customize_preview() ) {
-			return;
-		}
-		$handle = 'customize-snapshots-frontend';
-		wp_enqueue_script( $handle );
-
-		$exports = array(
-			'uuid' => $this->snapshot ? $this->snapshot->uuid() : null,
-			'home_url' => wp_parse_url( home_url( '/' ) ),
-			'l10n' => array(
-				'restoreSessionPrompt' => __( 'It seems you may have inadvertently navigated away from previewing a customized state. Would you like to restore the snapshot context?', 'customize-snapshots' ),
-			),
-		);
-		wp_add_inline_script(
-			$handle,
-			sprintf( 'CustomizeSnapshotsFrontend.init( %s )', wp_json_encode( $exports ) ),
+			$this->plugin->slug,
+			sprintf( 'new wp.customize.Snapshots( %s )', wp_json_encode( $exports ) ),
```

Comment:
This perfectly good and beautiful class is getting instantiated without saving a reference anywhere for us to use. It should get saved somewhere, like assigned to `wp.customize.snapshots`